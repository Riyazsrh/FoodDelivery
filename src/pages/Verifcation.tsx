import { Button, Container, Grid, ImageList, Typography,Box } from '@mui/material'
import { useStyles } from './VerifictionStyles'
import React, { useState } from 'react'
import { useNavigate, useParams } from 'react-router-dom'
import './Common.css';
import ReplayIcon from '@mui/icons-material/Replay';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import OtpInput from 'react-otp-input'

const Verifcation = () => {
  const { username } = useParams();

  const [password, setPassword] = useState<any>("");
  const [counter, setCounter] = React.useState(59);

  const classes = useStyles();
  const navigate = useNavigate();

  React.useEffect(() => {
    const timer: any =
      counter > 0 && setInterval(() => setCounter(counter - 1), 1000);
    return () => clearInterval(timer);
  }, [counter]);

  const handleChange = (event: any) => {
    setPassword(event);
    console.log(event);
  };
  const mobile = localStorage.getItem('mob');

  const userLogin = async () => {

    const Data: any = {
      username: username,
      password: password,
      device_token: "device token goes here",
      device_type: "android|ios|web",
    }

    await fetch("https://extended-retail-app.herokuapp.com/api/customer/login",
      {
        method: "POST",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",

        },
        body: JSON.stringify(Data),
      }
    )
      .then((resp) => resp.json())
      .then((respo) => {
        console.log("result", respo);
        if (respo.status === 200) {
          sessionStorage.setItem('logindata', JSON.stringify(respo))
          toast('Login Successful');
          navigate('/menu');
        } else {
          toast('Please Enter valid OTP');
        }
      });
  }

  return (
    <Container maxWidth="lg">
      <Grid container minHeight="100vh" direction="row" columns={{ xs: 4, sm: 10, md: 12 }} alignItems="center" justifyContent="center">
        <Grid item xs={5} sm={5} md={5} lg={5} >
          <Box sx={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
            <ImageList sx={{ ml: 10 }} >
              <img src='../images/logo 1.png' alt='logo' data-testid="img" />
            </ImageList>
            <Box >
              <Typography variant="h3" className={classes.verify} gutterBottom component="div">
                Verification
              </Typography>
            </Box>
            <Box >
              <Typography variant="h6" sx={{ display: "flex" }} className={classes.Login} gutterBottom component="div">
                Enter the OTP sent to +91 {mobile}
              </Typography>

              <Box sx={{ textAlign: "center" }}>
                <OtpInput
                  data-testid="otp-input"
                  placeholder="****"
                  inputStyle={{
                    width: "60px",
                    height: "50px",
                    borderRadius: "5px",
                    border: "1px solid #ccc",
                    margin: "0 5px 0 18px",
                    fontSize: "20px",
                    alignItems: "center"
                  }}
                  isInputNum
                  isInputSecure
                  value={password}
                  onChange={handleChange}
                  numInputs={4}
                />
                <Typography className={classes.timer}>Sec {counter} </Typography>
                <Box className="tog" >
                  <Typography className="resend" variant='h6' >
                    Resend OTP
                  </Typography>
                  <Button onClick={()=>{toast("OTP sent to your mobile number")}}> <ReplayIcon className="repay" /></Button>
                </Box>
              </Box>
            </Box>
            <Button data-testid="VerifcationButton" disabled={password.length < 4} className={password.length < 4 ? classes.VeriTxt1 : classes.VeriTxt} onClick={userLogin} >
              <Box className='btn-text'>Verifcation</Box>
            </Button>
          </Box>
        </Grid>

        <Grid item xs={7} sm={7} md={7} lg={7}>
          <Box>
            <img data-testid="mobImg" src='../images/img-1.png' alt='mobile' style={{ width: "100%", backgroundRepeat: "no-repeat" }} />
          </Box>
        </Grid>
      </Grid>
      <ToastContainer position="top-center" />
    </Container >
  )
}

export default Verifcation


