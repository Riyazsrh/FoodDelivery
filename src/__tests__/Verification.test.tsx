import { fireEvent, getByPlaceholderText, getByText, render, screen } from "@testing-library/react"
import Verifcation from "../pages/Verifcation"
import renderer from "react-test-renderer";

const mockUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUsedNavigate,
}));


test("Mobile image test", () => {
  render(<Verifcation />)
  const verify = screen.getByTestId("mobImg");
  expect(verify).toBeInTheDocument();
})


test("Verifcation img logo check", () => {
  const doc = render(<Verifcation />);
  const logo = doc.getByTestId('img');

  expect(logo).toHaveAttribute('src', './images/logo 1.png');
  expect(logo).toHaveAttribute('alt', 'logo');
})



it("should validate otp test", async () => {

  const { getByPlaceholderText } = render(<Verifcation />)

  fireEvent.change(getByPlaceholderText("otp"), {
    target: { value: "1234" }
  });

});

test("Verifcation button", () => {
  render(<Verifcation />)
  const button = screen.getByTestId("VerifcationButton")

  fireEvent.click(button);
  expect(button).toBeInTheDocument();

})

it("Verifcation snnapshot", () => {
  const tree = renderer.create(<Verifcation />)
  expect(tree).toMatchSnapshot();
})



jest.spyOn(window, 'alert').mockImplementation(() => { });