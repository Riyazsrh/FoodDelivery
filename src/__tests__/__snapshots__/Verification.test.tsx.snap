// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verifcation snnapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiGrid-root MuiGrid-container css-p08t7n-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 MuiGrid-grid-sm-5 MuiGrid-grid-md-5 MuiGrid-grid-lg-5 css-1qr3d4p-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-dvxtzn"
      >
        <ul
          className="MuiImageList-root MuiImageList-standard css-1xltci0-MuiImageList-root"
          style={
            Object {
              "gap": 4,
              "gridTemplateColumns": "repeat(2, 1fr)",
            }
          }
        >
          <img
            alt="logo"
            data-testid="img"
            src="./images/logo 1.png"
          />
        </ul>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom makeStyles-verify-29 css-1mktp8m-MuiTypography-root"
          >
            Verification
          </div>
        </div>
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom makeStyles-Login-30 css-1a1h3h8-MuiTypography-root"
          >
            Enter the OTP sent to +91 
          </div>
          <div
            className="MuiBox-root css-xi606m"
          >
            <div
              className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id=":r0:"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="otp"
                  required={false}
                  type="password"
                />
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-hdw1oc"
                  >
                    <span
                      className="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 makeStyles-timer-33 css-ahj2mt-MuiTypography-root"
            >
              Sec 
              59
               
            </p>
            <div
              className="MuiBox-root css-1h03bmi"
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <p
                className="MuiTypography-root MuiTypography-body1 makeStyles-ResendOtp-34 css-ahj2mt-MuiTypography-root"
                style={
                  Object {
                    "margin": "25px 3px",
                  }
                }
              >
                Resend OTP
              </p>
              <div
                className="bgg MuiBox-root css-0"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RefreshIcon"
                  focusable="false"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-VeriTxt-35 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="VerifcationButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="btn-text MuiBox-root css-0"
          >
            Verifcation
          </div>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 MuiGrid-grid-lg-7 css-1u5iry1-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-0"
      >
        <img
          alt="mobile"
          data-testid="mobImg"
          src="../images/img-1.png"
          style={
            Object {
              "backgroundRepeat": "no-repeat",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
